text/0000-geoip-plugins.md

```diff
@@ -0,0 +1,140 @@
+* Start Date: 2018-07-23
+* RFC PR: (leave this empty)
+* Fusion Issue: (leave this empty)
+
+## Summary
+
+Two new plugins, `fusion-plugin-geoip` and `fusion-plugin-geoip-react`. They
+can be used to geolocate the user based on their IP address and provide IP
+geolocation data to React components respectively.
+
+## Basic example
+
+### fusion-plugin-geoip
+
+```js
+const geoipData = GeoIP.from(ctx);
+console.log(geoipData.countryCode);
+console.log(geoipData.lat);
+console.log(geoipData.lng);
+```
+
+### fusion-plugin-geoip-react
+
+```js
+import {withGeoip} from 'fusion-plugin-geoip-react';
+
+function HelloGeoip({geoip}) {
+  return <p>Hello {geoip.countryName}</p>;
+}
+
+const HelloGeoipComponent = withGeoip(HelloGeoip);
+```
+
+## Motivation
+
+These plugins will support GeoIP, which is a common requirement for websites.
+For instance, GeoIP could be used to determine whether to show an EU cookie
+banner. Other potential use cases include CMS sites showing different content
+based on the user's country or map applications wanting to show a relevant
+section of the world to each user.
+
+The end goal is to have plugins that site builders can install, configure with
+a Maxmind database file, and immediately start using geoip in their Fusion
+application.
+	
+## Detailed design
+
+### fusion-plugin-geoip
+
+We will use `@lxe/maxmind-db-reader` to load the geoip data from `ctx.ip` on
```

Comment:
I think using this library has caused problems with the fusion build system since it imports a file from disk (or something like that, I don't recall the details but I think it causes our builds to print out "Warning: custom Babel config is an unstable API and may be not be supported in future releases. Use at your own risk.").

It seems like [node-geoip](https://github.com/bluesmoon/node-geoip) is the most popular JS library for interfacing with the maxmind database. Unfortunately this library requires transforming the database from the maxmind format but says it has better performance. I'm not sure if this is a good time to try migrating to this library.

Reply:
If there's a benchmark that demonstrates node-geoip is faster, I'm fine with using that. Otherwise, I think it should be OK to use maxmind-db-reader in the near term. We can fork it and/or send a PR upstream to fix the warning.